@anaprietonem I got most of the way there with this and then tried to validate configs — I now am getting 

```
💣 1 validation error for BaseSchema
diagnostics.log.mlflow
  Unable to extract tag using discriminator 'target_' | '_target_' [type=union_tag_not_found, input_value={'enabled': False, 'offli...ml_identity': 'managed'}, input_type=DictConfig]
    For further information visit https://errors.pydantic.dev/2.11/v/union_tag_not_found
[2025-10-13 13:17:53,712][anemoi.utils.cli][ERROR] - 💣 Exiting
```

when I try to do this. I assume this comes from the type annotation in the logging schema. Not too sure what to do about this, is that intended? Do I have to manually specify the logger `_target_`?

--- a/training/src/anemoi/training/diagnostics/mlflow/azureml.py
+++ b/training/src/anemoi/training/diagnostics/mlflow/azureml.py
@@ -105,8 +105,8 @@ def get_azure_workspace(
         msg = (
             "Azure environment incorrectly configured; tried to use \n  "
             f"- subscription: {sub}\n  - resource_group: {rg}\n  - workspace: {wsname}.\n"
-            "Try explicitly setting your subscription details via `diagnostics.mlflow.azureml.subscription_id`,"
-            "`diagnostics.mlflow.azureml.resource_group`, `diagnostics.mlflow.azureml.workspace_name`."
+            "Try explicitly setting your subscription details via `diagnostics.mlflow.subscription_id`,"
+            "`diagnostics.mlflow.resource_group`, `diagnostics.mlflow.workspace_name`."
         )
         raise ValueError(msg)
     LOGGER.info("Attempting to get Workspace object...")

--- a/training/src/anemoi/training/diagnostics/mlflow/logger.py
+++ b/training/src/anemoi/training/diagnostics/mlflow/logger.py
@@ -712,7 +712,6 @@ class BaseAnemoiMLflowLogger(MLFlowLogger, ABC):
         client: MlflowClient,
         run_id: str,
         params: dict[str, Any] | Namespace,
-        separate_configs: bool = False,
     ) -> None:
         """Log hyperparameters as an artifact."""
         import json
@@ -723,9 +722,8 @@ class BaseAnemoiMLflowLogger(MLFlowLogger, ABC):
             def default(self, o: Any) -> str:
                 return str(o)
 
-        now = str(datetime.datetime.now()).replace(" ", "T") if separate_configs else ""
         with tempfile.TemporaryDirectory() as tmp_dir:
-            path = Path(tmp_dir) / f"config.{now}.json"
+            path = Path(tmp_dir) / f"config.json"
             with Path.open(path, "w") as f:
                 json.dump(params, f, cls=StrEncoder)
             client.log_artifact(run_id=run_id, local_path=path)

--- a/training/src/anemoi/training/diagnostics/mlflow/logger.py
+++ b/training/src/anemoi/training/diagnostics/mlflow/logger.py
@@ -723,7 +723,7 @@ class BaseAnemoiMLflowLogger(MLFlowLogger, ABC):
                 return str(o)
 
         with tempfile.TemporaryDirectory() as tmp_dir:
-            path = Path(tmp_dir) / f"config.json"
+            path = Path(tmp_dir) / "config.json"
             with Path.open(path, "w") as f:
                 json.dump(params, f, cls=StrEncoder)
             client.log_artifact(run_id=run_id, local_path=path)

--- a/training/tests/unit/diagnostics/mlflow/test_azureml_mlflow_logger.py
+++ b/training/tests/unit/diagnostics/mlflow/test_azureml_mlflow_logger.py
@@ -4,6 +4,7 @@ import pytest
 
 from anemoi.training.diagnostics.mlflow.azureml import AnemoiAzureMLflowLogger
 from anemoi.training.diagnostics.mlflow.azureml import AzureIdentity
+from anemoi.training.schemas.diagnostics import AzureMlflowSchema
 
 
 @pytest.fixture(scope="session")
@@ -33,7 +34,7 @@ def default_logger(tmp_path, tmp_uri) -> AnemoiAzureMLflowLogger:
     return logger
 
 
-def test_mlflowlogger_params_limit(default_logger):
+def test_azure_mlflowlogger_params_limit(default_logger):
     default_logger._max_params_length = 3
     params = {"lr": 0.001, "path": "era5", "anemoi.version": 1.5, "bounding": True}
     # # Expect an exception when logging too many hyperparameters
@@ -41,10 +42,32 @@ def test_mlflowlogger_params_limit(default_logger):
         default_logger.log_hyperparams(params)
 
 
-def test_mlflowlogger_metric_deduplication(default_logger):
+def test_azure_mlflowlogger_metric_deduplication(default_logger):
     default_logger.log_metrics({"foo": 1.0}, step=5)
     default_logger.log_metrics({"foo": 1.0}, step=5)  # duplicate
     # Only the first metric should be logged
     assert len(default_logger._logged_metrics) == 1
     assert next(iter(default_logger._logged_metrics))[0] == "foo"  # key
     assert next(iter(default_logger._logged_metrics))[1] == 5  # step
+
+
+def test_azure_mlflow_schema():
+    config = {
+    '_target_':"anemoi.training.diagnostics.mlflow.azureml.AnemoiAzureMLflowLogger",
+    "enabled": False,
+    "offline": False,
+    "authentication": False,
+    "tracking_uri": None,  # You had ??? — using None as placeholder
+    "experiment_name": "anemoi-debug",
+    "project_name": "Anemoi",
+    "system": False,
+    "terminal": False,
+    "run_name": None,  # If set to null, the run name will be a random UUID
+    "on_resume_create_child": True,
+    "expand_hyperparams": ["config"],  # Which keys in hyperparams to expand
+    "http_max_retries": 35,
+    "max_params_length": 2000,
+    }
+    schema = AzureMlflowSchema(**config)
+
+    assert schema.target_ == "anemoi.training.diagnostics.mlflow.azureml.AnemoiAzureMLflowLogger"

